docs/06-contributors/999-rfcs/2023-04-16-website-resource.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Website resource
+description: How to implement the website resource
+---
+
+# Website resource
+
+- **Author(s):**: @tsuf239
+- **Submission Date**: 2023-04-16
+- **Stage**: draft
+- **Stage Date**: 2023-04-16
+
+<!---
+Directions for above:
+
+Author(s): Git tag for PR authors and contributors
+Submission Date: Fill in with date of initial submission, YYYY-MM-DD
+Stage: Fill in with current stage in the RFC lifecycle
+Stage Date: Fill in with date of last stage change
+-->
+
+> Implementing the website resource following the [SDK spec](../../04-reference/wingsdk-spec.md#website).
+
+<!--
+This RFCs have 3 sections: Requirements -> Design -> Implementation. We intentionally start with *Design* since it
+is a great way to introduice the feature to readers.
+-->
+
+## Implementation
+
+### Implementation for aws
+
+Following a few AWS guides on the internet (brought in the end of the section), it seems that the common way to serve a static website is by using a bucket.
+
+After creating a public bucket, we'll upload the folder from the relative path.
+AWS requires a index document name (to serve on the "/" path).
+We'll start by hardcoding it to be "index.html", and can add it later to the website props upon request.
+
+The last thing will be a registration to the chosen domain (if exists).
+
+The website resource supports preflight code only.
```

Comment:
I think being able to perform simple `get` requests to retrieve static files from the website could be a useful inflight api as well.

Reply:
In a "fetch" like operation? or "bucket.get" like?  if we'll reach it via "bucket.get" won't we lose the benefits of the caching via the CDN? 
It seems that an inflight fetch that is native to the language can be nice anyways :) 
(for limiting a common use of extern, and for this [issue](https://github.com/winglang/wing/issues/1850) as well )